Sources/Commands/PackageTools/InstalledPackages.swift

```diff
@@ -0,0 +1,122 @@
+            case 1:
+                productToInstall = possibleCanidates[0]
+            default: // More than one, check for possible
+                guard let product, let first = possibleCanidates.first(where: { $0.name == product }) else {
+                    throw StringError("Multiple canidates found, however, no product was specified. specify a product with the --product")
+                }
+                
+                productToInstall = first
+            }
+            
+            if let existingPkg = alreadyExisting.first(where: { $0.name == productToInstall.name }) {
+                throw StringError("\(productToInstall.name) is already installed at \(existingPkg.url)")
+            }
+            
+            try tool.createBuildSystem(explicitProduct: productToInstall.name).build(subset: .product(productToInstall.name))
+            
+            let binPath = try tool.buildParameters().buildPath.appending(component: productToInstall.name)
+            let finalBinPath = swiftpmBinDir.appending(component: binPath.basename)
+            try tool.fileSystem.copy(from: binPath, to: finalBinPath)
+            
+            let pkgInstance = InstalledPackage(name: productToInstall.name, url: .init(finalBinPath))
+            alreadyExisting.append(pkgInstance)
+            
+            try JSONEncoder().encode(path: installedPackagesJSONPath, fileSystem: tool.fileSystem, alreadyExisting)
+        }
+    }
+    
+    struct Remove: SwiftCommand {
+        @OptionGroup
+        var globalOptions: GlobalOptions
+        
+        @Option(help: "Name of the executable to remove.")
+        var name: String
+        
+        func run(_ tool: SwiftTool) throws {
+            let dotSwiftPMDir = try tool.fileSystem.dotSwiftPM
+            let registryURL = dotSwiftPMDir.appending(component: "installedPackages.json").asURL
+            var alreadyRegistered = (try? InstalledPackage.registered(registryPath: .init(try tool.fileSystem.getOrCreateSwiftPMInstalledBinariesDirectory()),
+                                                                      withFS: tool.fileSystem)) ?? []
+            
+            guard let whatWeWantToRemove = alreadyRegistered.first(where: { $0.name == name || $0.url.basename == name }) else {
+                throw StringError("No such installed executable as \(name)")
+            }
+            
+            try tool.fileSystem.removeFileTree(whatWeWantToRemove.url)
+//            try FileManager.default.removeItem(at: whatWeWantToRemove.url)
+            alreadyRegistered.removeAll(where: { $0 == whatWeWantToRemove })
+            try JSONEncoder().encode(alreadyRegistered).write(to: registryURL)
+            print("Removed \(name).")
+        }
+    }
+}
+
+fileprivate struct InstalledPackage: Codable, Equatable {
+    
+    static func registered(registryPath: AbsolutePath, withFS fileSystem: FileSystem) throws -> [InstalledPackage] {
```

Comment:
`withFS` feels redundant as a parameter label as we can only pass filesystems as arguments anyway
```suggestion
    static func registered(registryPath: AbsolutePath, _ fileSystem: FileSystem) throws -> [InstalledPackage] {
```